### PR TITLE
Support expected-stdout, expected-stderr by default

### PR DESCRIPTION
**Pull request by michaljurecko**
Solving: https://keboola.atlassian.net/browse/COM-203

**Problem 1:**
- V komponentach casto vyuzivame datadir-testy, no potrebujeme kontrolovat aj `stdout` (pri sync akciach) a `stderr` (pri chybach)
- Podpora pre `stdout` a `stderr` je implementovana v `DatadirTestSpecification`, ale chybala defaultna implementacia `expected-stderr` a `expected-stdout`.
- Teda v kazdom projekte sme doteraz museli prepisovat metodu `processOneTest`:
   - napr: [tu](https://github.com/keboola/synapse-transformation/blob/5864529ff4acbc6e624925f919404e5659e0bff6/tests/functional/DatadirTestsProvider.php#L49), [tu](https://github.com/keboola/ex-onedrive/blob/fe86a1db893ba295bb6e76be3e936f3e54d600c6/tests/datadir/DatadirTestsProvider.php#L51), [tu](https://github.com/keboola/db-extractor-informix/blob/f43de64ec06268a3072c1118b18c3583a41e45ea/tests/functional/DatadirTestsProvider.php#L51) .....

**Problem 2:**
- `stdout` a `stderr` potrebujeme porovnavat cez `assertStringMatchesFormat` nie `assertSame`
- vo vystupe sa skor vzdy objavi nejake IDcko, ktore je pri kazdom behu ine
- alebo nejaky prefix, ktory je závislý na verzii databazy a testujeme to oproti viacerym verziam, a pod.
- ... ak sa pouzije `assertStringMatchesFormat`, tak v `expected-stderr` mozeme pouzit placeholdery, napr `%a`.

### CODE REVIEW

File: src/AbstractDatadirTestCase.php [hunk 1]
```diff
@@ -138,16 +138,16 @@ protected function assertMatchesSpecification(
             $this->assertNotSame(0, $runProcess->getExitCode(), 'Exit code should have been non-zero');
         }
         if ($specification->getExpectedStdout() !== null) {
-            $this->assertSame(
-                $specification->getExpectedStdout(),
-                $runProcess->getOutput(),
+            $this->assertStringMatchesFormat(
```
Comment (michaljurecko) [inline src/AbstractDatadirTestCase.php:141]: `stdout` a `stderr` porovnavame cez **`assertStringMatchesFormat`**.
Okrem toho je tu pridany aj **`trim`**, kedze synchronne akcie casto neobsahuju `\n` na konci vystupu a IDEcko obycajne prida `\n` do `expected-stdout` a potom sa to tazko debuguje, preco to spadlo.

**Comment by michaljurecko** [comment]
@zajca riesil uz skor `stderr` ale z nejakeho dovodu sa to nemerglo:
https://github.com/keboola/datadir-tests/pull/21

**Comment by ondrajodas** [comment]
cca za hoďku na to kouknu

**Review by ondrajodas — APPROVED** [review]
lgtm

**Pull request merged**